server/cssfile.js

```diff
@@ -35,42 +35,64 @@ CssFile.prototype.selectorFromPosition = function(line, column){
 };
 
 CssFile.prototype.setContent = function(source, callback){
-	var messages = csslint.verify(source).messages;
-	var errors = [];
-	messages.forEach(function(msg){
-		if(msg.type == 'error'){
-			errors.push(msg);
+	const errors = [];
+	this.source = source;
+	const lint = stylelint.lint({code: source}).then((result) => result.results).then(results => {
+		results.forEach((msg) =>{
+			if(msg.errored){
+				msg.warnings.forEach(warning => {
+					errors.push(warning)
+				});
+			}
+		})
+		return results
+	}).then(results => {
+		if(errors.length > 0 && callback){
+			throw new Error("Errors found")
 		}
-	});
+		return results
+	}).catch(err => {
+		return err
+	})
 
-	if(errors.length > 0 && callback){
-		callback(errors);
-		return;
+	const parser =  new Promise((res, rej) => { 
+		try {
+			const parsed = cssparser.parse(source)
+			this.parsed = parsed
+			res(parsed)
+		}
+		catch(err){
+			rej(err);}
 	}
+	).then((parsed) => {
+		for (const rule of parsed.nodes) {
+			let source = rule.source;
+			source.start.line--;
+			source.start.column--;
+			source.end.column++;
+		}
+	}).catch(err => {
+		return err
+	})
 
-	var changed = (this.source != undefined && this.source != source);
-
-	this.source = source;
-
-	try{
-		this.parsed = cssparser.parse(source);
-	}catch(err){
-		callback(err);
-		return;
+	Promise.all([lint, parser]).then(result => {
+		var changed = (this.source != undefined && this.source != source);
+		if(changed){
+			callback(null);
+		}else{
+			callback(null, null);
+		}
 	}
+	).catch(err => {
```

Comment:
my understanding of js promises is this `.catch` should never be called since the error will already be caught in each of promise `.catch`es above. I'm suspicious this is a fluke of the test suite. What's actually happening here is the callback on line 80/83 is being called incorrectly during the test which throws an error from `expect(err).to.not.be.null` which then leads to this `.catch` and seemingly correct behavior. You can fix the test suite by having it assert the error value and not just the fact that it errored.